components/Auctions/index.js

```diff
@@ -0,0 +1,156 @@
+import Image from 'next/image';
+import React, { useState, useEffect } from 'react';
+import styles from './Auctions.module.css';
+import fetchData from '../../utils/fetchData';
+import fetchSelfDetails from '../../utils/fetchSelfDetails';
+
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const AUCTIONS_URL = `${BASE_API_URL}/auctions`;
+const BASE_IMAGE_URL =
+  'https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main/members';
+
+const HandleAuctions = () => {
+  const [auctionsData, setAuctionsData] = useState([]);
+  const [userBid, setUserBid] = useState();
+  const [isUserLoggedIn, setIsUserLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetchAndSetAuctions();
+  }, []);
+
+  useEffect(() => {
+    fetchSelfDetails()
+      .then((res) => {
+        if (res.status === 200) {
+          setIsUserLoggedIn(true);
+        }
+      })
+      .catch((err) => {
+        console.log('User is not logged in', err);
+      });
+  }, []);
+
+  const fetchAndSetAuctions = async () => {
+    const response = await fetchData(`${AUCTIONS_URL}`);
+    const json = await response.json();
+    setAuctionsData(json.auctions);
+    setIsLoading(false);
+  };
+
+  const handleNewBid = async (e, auctionId) => {
+    if (!isUserLoggedIn) {
+      return alert('Please log in to bid!');
+    }
+    setIsLoading(true);
+    e.preventDefault();
+    const reqBody = { bid: userBid };
+    const response = await fetchData(
+      `${AUCTIONS_URL}/bid/${auctionId}`,
+      'POST',
+      {
+        credentials: 'include',
+        body: JSON.stringify(reqBody),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    );
+    const { status } = await response;
+    if (status === 204) {
+      fetchAndSetAuctions();
+      setIsLoading(false);
+    }
+  };
+
+  const validateBid = (userBid, highestBid) => {
+    if (userBid <= highestBid) alert('You must bid higher than current bid!');
+    else setUserBid(userBid);
+  };
+
+  const getColumns = (totalBidder) => {
+    return totalBidder <= 12 ? Math.ceil(totalBidder / 2) : 7;
```

Comment:
What are 12 and 7?

Reply:
MAX_ALLOWED_COLUMNS=7
MAX_BIDDERS_ALLOWED_BELOW_MAX_COLUMNS=12